pkg/test/utils/kubernetes.go

```diff
@@ -774,3 +777,105 @@ func StartTestEnvironment() (env TestEnvironment, err error) {
 	env.DiscoveryClient, err = discovery.NewDiscoveryClientForConfig(env.Config)
 	return
 }
+
+// Kubectl runs a kubectl command (or plugin) with args.
+// args gets split on spaces (respecting quoted strings).
+func Kubectl(ctx context.Context, namespace string, args string, cwd string, stdout io.Writer, stderr io.Writer) error {
+	argSlice := []string{
+		"kubectl",
+	}
+
+	argSplit, err := shlex.Split(args)
+	if err != nil {
+		return err
+	}
+
+	argSlice = append(argSlice, argSplit...)
+	argSlice = append(argSlice, "--namespace", namespace)
+
+	actualDir, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+
+	cmd := exec.Command("kubectl")
+	cmd.Args = argSlice
+	cmd.Dir = cwd
+	cmd.Stdout = stdout
+	cmd.Stderr = stderr
+	cmd.Env = []string{
+		fmt.Sprintf("KUBECONFIG=%s/kubeconfig", actualDir),
+		fmt.Sprintf("PATH=%s/bin/:%s", actualDir, os.Getenv("PATH")),
+	}
+
+	return cmd.Run()
+}
+
+// Kubeconfig converts a rest.Config into a YAML kubeconfig and writes it to w
+func Kubeconfig(cfg *rest.Config, w io.Writer) error {
```

Comment:
I wasn't able to find a method that could convert a rest.Config to a kubeconfig already. There's a number of cases that we need to be able to support:

* Merged kubeconfig
* In cluster config
* Mocked control plane (just requires setting the Server setting)
* Kind config

In some cases we have the path and could copy it.. but other times the path might not exist or be ambiguous and we would want to generate a kubeconfig.

But maybe there's a better way to do this.. Thoughts?